### PR TITLE
feat: Add opendal hello world example

### PR DESCRIPTION
**Pull request by Xuanwo**
## Description of change

Add a hello world example for opendal

## How has this been tested? (if applicable)

Local tested.

### CODE REVIEW

**Comment by jonaro00** [comment]
Thanks!
Since we are for now mainly organising by web framework (the structure will improve later), can you
1. Move it to `rocket/opendal`
2. add a line in the README.md

**Comment by Xuanwo** [comment]
Got it. I want to extend the examples to s3/azblob and so on. Do you have suggestions for the layout?

**Comment by jonaro00** [comment]
Aren't those examples quite trivial when compared against the in-memory one? Wouldn't they be identical but with an added Secrets.toml.example file and a different scheme param?
Showing how to do it for S3 in `rocket/opendal-s3` should be sufficient for showing the pattern.

You could add

**Comment by Xuanwo** [comment]
cc @jonaro00, PTAL, thanks!

File: rocket/opendal-memory/README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# OpenDAL Memory Storage
+
+This example shows that how to connect to an in-memory storage using OpenDAL.
+
+## Project structure
+
+The project consists of the following files
+
+- `Shuttle.toml` contains the name of the app
+- `src/lib.rs` is where all the magic happens - it creates a shuttle service with two endpoints: one for add new data and one for retrieving it back.
```
Comment (jonaro00) [inline rocket/opendal-memory/README.md:10]:
```suggestion
- `src/main.rs` is where all the magic happens - it creates a Shuttle service with two endpoints: one for adding new data and one for retrieving it back.
```

**Pull request merged**